packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -68,6 +68,11 @@ abstract class BuildFrameworkCommand extends BuildSubCommand {
       ..addFlag('cocoapods',
         help: 'Produce a Flutter.podspec instead of an engine Flutter.xcframework (recommended if host app uses CocoaPods).',
       )
+      ..addFlag('plugins',
+        defaultsTo: true,
+        help: 'Whether to produce frameworks for the plugins. '
+              'Consider using "--no-plugins" if plugin frameworks are not required.',
+      )
       ..addFlag('static',
```

Comment:
Passing both `--static` and `--no-plugins` should be an error, since that would be a mistake, and it's not clear what someone would expect it to do.

Reply:
Added this to validations